facebook/facebook-json.c

```diff
@@ -60,8 +62,15 @@ json_value *fb_json_new(const gchar *data, gsize length, GError **err)
         return json;
     }
 
+    /* Ensure it's null-terminated before passing it to g_strescape() */
+    dstr = g_strndup(data, MIN(length, 400));
```

Comment:
Why not just remove this duplication all together?  For one, this is the JSON interface, not the IRC interface.  Second, since there is no *real* need for this, aside from doubly duplicating the string.